File: Palace/OPDS/TPPOPDSType.m
#import "TPPOPDSType.h"

BOOL TPPOPDSTypeStringIsAcquisition(NSString *const string)
{
  return string != nil && [string rangeOfString:@"acquisition"
                       options:NSCaseInsensitiveSearch].location != NSNotFound;

}

BOOL TPPOPDSTypeStringIsNavigation(NSString *const string)
{
  return string != nil && [string rangeOfString:@"navigation"
                       options:NSCaseInsensitiveSearch].location != NSNotFound;

}

BOOL TPPOPDSTypeStringIsOpenSearchDescription(NSString *string)
{
  return string != nil && [string rangeOfString:@"application/opensearchdescription+xml"
                       options:NSCaseInsensitiveSearch].location != NSNotFound;

}

File: Palace/OPDS/TPPOPDSType.swift
import Foundation

func TPPOPDSTypeStringIsAcquisition(_ string: String?) -> Bool {
  guard let string = string else { return false }
  return string.range(of: "acquisition", options: .caseInsensitive) != nil
}

func TPPOPDSTypeStringIsNavigation(_ string: String?) -> Bool {
  guard let string = string else { return false }
  return string.range(of: "navigation", options: .caseInsensitive) != nil
}

func TPPOPDSTypeStringIsOpenSearchDescription(_ string: String?) -> Bool {
  guard let string = string else { return false }
  return string.range(of: "application/opensearchdescription+xml", options: .caseInsensitive) != nil
}
